src/pages/Library.tsx

```diff
@@ -157,15 +227,316 @@ class Library extends React.Component<ILibraryProps, ILibraryState>
             scrollStamp: e.timeStamp
         });
     }
-    //Change the chapter shown to new chapter of curNum.
-    setChapter(chapter: any){
-        console.log(this.state.context);
+
+    
+    //Track scroll to bottom
+    onIonScroll=  (ev : any) => {
+        const elem = document.querySelector("ion-content")!.getScrollElement();
+        // the ion content has its own associated scrollElement
+        //elem > scrollHeight = 9090 by default.
+        var show = false;
+        const retVal = elem.then((result) =>{
+            const h = result.scrollHeight
+            const pos = result.scrollTop
+            
+            //Control fixed arrows' appearances: if down-scroll, appear.
+            var areArrowsShown = false;
+            console.log(this.state.prevHeight);
+            console.log(pos);
+            if(this.state.prevHeight > pos){ 
+               
+                areArrowsShown = true;
+                this.setState(update(this.state,
+                    {prevHeight: {$set : pos}
+                }));
+                
+            }else{
+                areArrowsShown = false;
+                this.setState(update(this.state,
+                    {prevHeight: {$set : pos}
+                }));
+            }
+            
+            //We can never truly get to the bottom bc container doesn't take up 100% of screen.
+            //so, estimate getting to the bottom.... i wish there were a better way to do this
+            if(h <= pos + 800){ 
+
+                this.setState({showArrows: false, showFab: true})
```

Comment:
This is the right way to use setState, why were you using it differently before?